diff --git a/sdk/foobar2000/foobar2000_component_client/component_client.cpp b/sdk/foobar2000/foobar2000_component_client/component_client.cpp
index 7ad217a..d7a1017 100644
--- a/sdk/foobar2000/foobar2000_component_client/component_client.cpp
+++ b/sdk/foobar2000/foobar2000_component_client/component_client.cpp
@@ -107,7 +107,7 @@ namespace {
 		}
 
 		bool is_debug() override {
-			return PFC_DEBUG != 0;
+			return false;
 		}
 	};
 }
