diff --git a/CMakeLists.txt b/CMakeLists.txt
index a64ca73..e76a6d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,17 +82,12 @@ CHECK_TYPE_SIZE("short" SIZEOF_SHORT)
 
 TEST_BIG_ENDIAN(HOST_BIG_ENDIAN)
 
-check_c_compiler_flag(-fvisibility=hidden EVHTP_HAS_VISIBILITY_HIDDEN)
 check_c_compiler_flag(-std=c99 EVHTP_HAS_C99)
 
 if (EVHTP_HAS_C99)
 	add_definitions(-DEVHTP_HAS_C99)
 endif()
 
-if (EVHTP_HAS_VISIBILITY_HIDDEN)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden")
-endif()
-
 if (NOT HAVE_SYS_TREE)
     CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/compat/sys/tree.h.in
 			${CMAKE_CURRENT_BINARY_DIR}/compat/sys/tree.h)
diff --git a/evhtp-config.h.in b/evhtp-config.h.in
index d6ddbc7..574506e 100644
--- a/evhtp-config.h.in
+++ b/evhtp-config.h.in
@@ -6,14 +6,9 @@ extern "C" {
 #endif
 
 #ifndef EVHTP_EXPORT
-# if (defined __GNUC__ && __GNUC__ >= 4) || defined __INTEL_COMPILER || defined __clang__
-#  define EVHTP_EXPORT __attribute__ ((visibility("default")))
-# else
-#  define EVHTP_EXPORT
-# endif
+#define EVHTP_EXPORT
 #endif
 
-
 #undef EVHTP_DISABLE_EVTHR
 #undef EVHTP_DISABLE_REGEX
 #undef EVHTP_DISABLE_SSL
